metals/src/main/scala/scala/meta/internal/metals/QuickFixes.scala

```diff
@@ -0,0 +1,82 @@
+package scala.meta.internal.metals
+
+import scala.concurrent.Future
+import scala.meta.pc.CancelToken
+import org.eclipse.{lsp4j => l}
+import scala.concurrent.ExecutionContext
+import scala.meta.internal.metals.MetalsEnrichments._
+
+trait QuickFixes {
+  trait QuickFix {
+    def contribute(implicit ec: ExecutionContext): Future[Seq[l.CodeAction]]
```

Comment:
I wonder if we should have contribute returning and `Option[CodeAction]`, we could have a lsit of quickfixes and just flatMap over them to get the code actions.

`contribute` would also take other parameters from below that we could need in any quickFix code action.

Reply:
The issue is that some computations are asynchronous (hence `Future`) and there may be different imports available for an unknown symbol (hence `Seq`).

Or did I not understand your proposal?

Reply:
Or just the Seq would actually also work:
```scala
trait QuickFix {
    def contribute( 
        diagnostics: Seq[l.Diagnostic],
        params: l.CodeActionParams,
        compilers: Compilers,
        token: CancelToken)
   (implicit ec: ExecutionContext): Future[Seq[l.CodeAction]]
}
```

and then we could do:

```scala
List(ImportMissingSymbol, Other).flatMap(_.contribute)
```
It would be easier to keep the actions separate in their own files.

Reply:
@gabro Sure it should be with futures with Seq. Let me fix the examples a bit.

Reply:
I just meant it would be useful to have an interface describing what you can use in the quickfix and just add it to the list.

Reply:
ahh, I see. That might work, I'll have a try! :)

Reply:
Take a look now, I think it's close to what you mean @tgodzik